pkg/controller/installation/products/codeready/reconciler.go

```diff
@@ -526,26 +464,24 @@ func (r *Reconciler) reconcileBlackboxTargets(ctx context.Context, inst *v1alpha
 	err = monitoring.CreateBlackboxTarget("integreatly-codeready", v1alpha12.BlackboxtargetData{
 		Url:     r.Config.GetHost(),
 		Service: "codeready-ui",
-	}, ctx, cfg, inst, client)
+	}, ctx, cfg, r.installation, client)
 	if err != nil {
 		return v1alpha1.PhaseFailed, errors.Wrap(err, "error creating codeready blackbox target")
 	}
 
 	return v1alpha1.PhaseCompleted, nil
 }
 
-func (r *Reconciler) createCheCluster(ctx context.Context, kcCfg *config.RHSSO, kr *keycloakv1.KeycloakRealm, inst *v1alpha1.Installation, serverClient pkgclient.Client) (*chev1.CheCluster, error) {
-	selfSignedCerts := inst.Spec.SelfSignedCerts
+func (r *Reconciler) createCheCluster(ctx context.Context, kcCfg *config.RHSSO, kr *keycloakv1.KeycloakRealm, serverClient pkgclient.Client) (*chev1.CheCluster, error) {
+	selfSignedCerts := r.installation.Spec.SelfSignedCerts
 
-	// setup external postgres db using cloud-resources-operator default is Openshift
-	cheClusterExternalPostgres, err := r.reconcileExternalPostgres(ctx, inst, serverClient)
+	// get the postgres cloud resources operator cr
+	croSec := &v1.Secret{}
+	postgresName := fmt.Sprintf("codeready-postgres-%s", r.installation.Name)
+	err := serverClient.Get(ctx, pkgclient.ObjectKey{Name: postgresName, Namespace: r.installation.Namespace}, croSec)
```

Comment:
I think it would be better to first get the postgres custom resource and then get the secret from the secret ref, since the secret name and namespace might differ depending on the cr configuration. 

```suggestion
	err := serverClient.Get(ctx, pkgclient.ObjectKey{Name: postgres.Status.SecretRef.Name, Namespace: postgres.Status.SecretRef.Namespace}, croSec)
```